acceptance/go.mod

```diff
@@ -1,10 +1,19 @@
 module github.com/hashicorp/consul-k8s/acceptance
 
-go 1.20
+go 1.21.1
+
+toolchain go1.22.0
+
+replace (
+	k8s.io/api => k8s.io/api v0.26.4
+	k8s.io/apimachinery => k8s.io/apimachinery v0.26.4
+	k8s.io/client-go => k8s.io/client-go v0.26.4
+	sigs.k8s.io/controller-runtime => sigs.k8s.io/controller-runtime v0.14.6
```

Comment:
is there a reason we need to have the replace statements?

Reply:
Yes, by upgrading this library `github.com/gruntwork-io/terratest v0.31.2` -> `v0.46.7`, the other dependencies were upgraded to version `k8s.io/apimachinery v0.28.4` resulting in this known [error](https://github.com/operator-framework/operator-sdk/issues/6396)
```
# sigs.k8s.io/controller-runtime/pkg/cache
../../../go/pkg/mod/sigs.k8s.io/controller-runtime@v0.14.7/pkg/cache/multi_namespace_cache.go:308:9: cannot use handles (variable of type map[string]"k8s.io/client-go/tools/cache".ResourceEventHandlerRegistration) as "k8s.io/client-go/tools/cache".ResourceEventHandlerRegistration value in return statement: 
map[string]"k8s.io/client-go/tools/cache".ResourceEventHandlerRegistration does not implement "k8s.io/client-go/tools/cache".ResourceEventHandlerRegistration (missing method HasSynced)
FAIL	github.com/hashicorp/consul-k8s/acceptance/tests/api-gateway [build failed]
```
 
The only [solution](https://github.com/operator-framework/operator-sdk/issues/6396) currently is to downgrade the versions of the dependencies OR not upgrade `github.com/gruntwork-io/terratest` library. So i picked the latter